### PR TITLE
[Term Entry] Dart Type Conversion .tryParse() #4572

### LINKED ISSUE #4572: [Term Entry] Dart Type Conversion .tryParse()
### Reason/inspiration (optional)

We would like a new term entry under Dart for the Dart Type Conversion `.tryParse()` method.
Provide the entry in a markdown file under docs/content/dart/concepts/type-conversion/tryParse/tryParse.md

The entry should
- Have an introductory paragraph about what the method is and does.
- A syntax section with a pseudo code block with an explanation of it.
- An example showing the method in use.

Please refer to the [entry templates](https://github.com/Codecademy/docs/blob/main/documentation/), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by lou-pastorino**
### Description

 - Added the tryParse() term for Dart. 
 - Mardown file entry is under: docs/tree/term-tryParse/content/dart/concepts/type-conversion 

### Issue Solved

- Closes #4572

### Type of Change

- Adding a new entry

### Checklist

- [x ] All writings are my own.
- [ x] My entry follows the Codecademy Docs style guide.
- [x ] My changes generate no new warnings.
- [ x] I have performed a self-review of my own writing and code.
- [ x] I have checked my entry and corrected any misspellings.
- [ x] I have made corresponding changes to the documentation if needed.
- [ x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [ x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [ x] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

**Comment by mamtawardhani** [comment]
Hey @lou-pastorino! Please update the path to `docs/content/dart/concepts/type-conversion/terms/tryParse/tryParse.md`

File: content/dart/concepts/type-conversion/terms/tryParse/tryParse.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+In Dart, .tryParse() is a static method available on several data types, such as int, for example. The .tryParse() method returns a nullable value of the target data type if the parsing is successful. If parsing fails, the returned value is null.  
+
+## Syntax
+
+```pseudo
+dataType.tryParse(stringValue)
+```
+1. stringValue is the string one wishes to convert into a specific data type.
+2. dataType is the data type that one wishes to convert stringValue to.
```
Comment (mamtawardhani) [inline content/dart/concepts/type-conversion/terms/tryParse/tryParse.md:25]:
```suggestion
- `stringToParse` - Represents the `string` to be converted into the desired data type.
- `dataType` - Represents the data type to be converted to stringValue.
```

File: content/dart/concepts/type-conversion/terms/tryParse/tryParse.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+ - 'learn-dart'
+  - 'paths/computer-science'
+---
+
+In Dart, .tryParse() is a static method available on several data types, such as int, for example. The .tryParse() method returns a nullable value of the target data type if the parsing is successful. If parsing fails, the returned value is null.  
+
+## Syntax
+
+```pseudo
+dataType.tryParse(stringValue)
```
Comment (mamtawardhani) [inline content/dart/concepts/type-conversion/terms/tryParse/tryParse.md:22]:
```suggestion
dataType.tryParse(stringToParse)
```

File: content/dart/concepts/type-conversion/terms/tryParse/tryParse.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  } else {
+    print('Failed to parse the string as an integer');
+  }
+}
+```
+The above code will return the following output: 
+
+```dart
+//Parsed integer: 42
+```
```
Comment (mamtawardhani) [inline content/dart/concepts/type-conversion/terms/tryParse/tryParse.md:51]: The ## Codebyte is not needed, please place this codebyte under ## Example along with its output. Remove the:
value = num.tryParse('5'); // 5
2. value = num.tryParse('five'); // null

and ## Codebyte example tag

File: content/dart/concepts/type-conversion/terms/tryParse/tryParse.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+
```
Comment (mamtawardhani) [inline content/dart/concepts/type-conversion/terms/tryParse/tryParse.md:53]:
```suggestion
```

**Review by mamtawardhani — CHANGES_REQUESTED** [review]
Hey @lou-pastorino!
Thanks for contributing to the Codecademy docs! 

I've suggested a few changes, please review and modify them at your earliest convenience.
Thank you!

File: content/dart/concepts/type-conversion/terms/tryParse/tryParse.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+In Dart, **`.tryParse()`** is a static method available on several data types, such as `int` and `double`, for example. When called on a `string`, the `.tryParse()` method attempts to parse the string and returns a nullable value of the target data type if the parsing is successful. However, if parsing fails due to an invalid format in the string, the method returns `null`.
+
+## Syntax
+
+```pseudo
+dataType.tryParse(stringToParse)
+```
+- `stringToParse` - Represents the `string` to be converted into the desired data type.
+- `dataType` - Represents the data type to be converted to stringValue.
```
Comment (mamtawardhani) [inline content/dart/concepts/type-conversion/terms/tryParse/tryParse.md:27]:
```suggestion
- `stringToParse` - Represents the `string` to be converted into the desired data type.
- `dataType` -  Represents the data type used to parse `stringToParse`.
```

**Review by mamtawardhani — CHANGES_REQUESTED** [review]
Hey @lou-pastorino!

Can you please implement the one suggestion given? Thanks!

**Comment by lou-pastorino** [comment]
@mamtawardhani Hey! The final suggestion has been implemented. 
Thanks

**Comment by mamtawardhani** [comment]
Hey @lou-pastorino! Thank you for implementing the suggestions!

Lastly, could you please update the path to `docs/content/dart/concepts/type-conversion/terms/tryParse/tryParse.md`

**Comment by lou-pastorino** [comment]
> Hey @lou-pastorino! Thank you for implementing the suggestions!
> 
> Lastly, could you please update the path to `docs/content/dart/concepts/type-conversion/terms/tryParse/tryParse.md`

Hi @mamtawardhani 

Apologies, I must be getting confused somewhere. On my side it looks like the file path is as you have described above. Do I need to create a new pull request to merge the updated branch with my main branch to update the file path? 
Apologies for the confusion!

**Review by mamtawardhani — APPROVED** [review]
Good to go for the second review.

@lou-pastorino Thanks for making the changes!

**Pull request merged**